ecommerce/src/main/java/com/algolia/instantsearch/examples/ecommerce/EcommerceActivity.java

```diff
@@ -96,4 +180,25 @@ private void toggleArrow(Button b, boolean up) {
         b.setCompoundDrawablesWithIntrinsicBounds(currentDrawables[0], currentDrawables[1], newDrawable, currentDrawables[3]);
 
     }
+
+    private void showPermissionOverlay() {
```

Comment:
nit: Can we put those next 2 functions after the `onTapMic` that uses them? would be easier when reading code